drivers/modem/modem_iface_uart.c

```diff
@@ -99,6 +99,11 @@ static void modem_iface_uart_isr(const struct device *uart_dev,
 	if (total_size > 0) {
 		k_sem_give(&data->rx_sem);
 	}
+
+	/* Disable the TX interrupt once tranmission is complete */
+	if (uart_irq_tx_complete(ctx->iface.dev) > 0) {
+		uart_irq_tx_disable(ctx->iface.dev);
```

Comment:
The TX interrupt does not seem to be enabled by this driver (it sends data using `uart_poll_out`), so I don't entirely understand how is this supposed to help. What am I missing here?

Reply:
Is there a chance that the Nordic UARTE driver has a bug that triggers the TX interrupt even when not using interrupt-driven transmission?

Reply:
That was my initial thought, but I couldn't find in the UARTE driver code anything suspicious in this regard, hence I added my above comment. Now I additionally made a simple test application to check the driver behavior and it seems to work fine with RX using interrupts and TX only calling `uart_poll_out`.

Reply:
We use the HL7800 modem driver on our product with an NRF52840 and do not see any issues like this.
The HL7800 uses `modem_recevier.c` and `mdm_receiver_setup` disables the TX interrupt before any other functions get used. This could be why it works ok in our case.

Reply:
Could it be that `CONFIG_UART_<n>_ENHANCED_POLL_OUT=n`? If this option is disable then there is TX interrupt on ENDTX event just to trigger STOPTX task.

Reply:
I set up the hardware again and the `TXSTOPPED` interrupt is definitely being enabled through `uart_poll_out`.
https://github.com/zephyrproject-rtos/zephyr/blob/83939c267c1a4ffe2644a913da79b5409e114f81/drivers/serial/uart_nrfx_uarte.c#L1493
https://github.com/zephyrproject-rtos/zephyr/blob/83939c267c1a4ffe2644a913da79b5409e114f81/drivers/serial/uart_nrfx_uarte.c#L597
Full call stack:
![image](https://user-images.githubusercontent.com/8476715/152121932-4ebaddb7-6ec2-4e5f-8386-7bc76cfb49ed.png)